Add payment details display to Link inline 2FA section.

carlosmuvi-stripe opened the pull request:
# Summary
<!-- Simple summary of what was changed. -->

# Motivation
<!-- Why are you making this change? If it's for fixing a bug, if possible, please include a code snippet or example project that demonstrates the issue. -->

# Testing
<!-- How was the code tested? Be as specific as possible. -->
- [ ] Added tests
- [ ] Modified tests
- [ ] Manually verified

<!-- Ignored Tests Did you newly ignore a test in this PR?  If so, please open an R4 incident so that the test can be re-enabled as soon as possible-->

# Screenshots
| Before  | After |
| ------------- | ------------- |
| *before screenshot*  | *after screenshot* |

# Changelog
<!-- Is this a notable change that affects users? If so, add a line to `CHANGELOG.md` and prefix the line with one of the following:
    - [Added] for new features.
    - [Changed] for changes in existing functionality.
    - [Deprecated] for soon-to-be removed features.
    - [Removed] for now removed features.
    - [Fixed] for any bug fixes.
    - [Security] in case of vulnerabilities.
-->

lng-stripe commented on paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/DefaultPaymentUI.kt:
```diff
@@ -0,0 +1,43 @@
+
+    val paymentIcon = when (defaultPaymentType) {
+        "CARD" -> CardBrand.fromCode(defaultCardBrand).getCardBrandIconForVerticalMode()
+        "BANK_ACCOUNT" -> R.drawable.stripe_link_bank_outlined
```
I don't think the bank icon will look right -- we have special casing for bank account: https://github.com/stripe/stripe-android/blob/master/paymentsheet/src/main/java/com/stripe/android/link/ui/wallet/PaymentDetails.kt#L431-L442

lng-stripe commented on paymentsheet/src/test/snapshots/images/com.stripe.android.link.ui.wallet_LinkInline2FASectionScreenshotTest_testWithPaymentDetailsMastercard[LightTheme,LargeFont].png:
🤔 It's kind of weird for the last4 to scale while the Link logo doesn't. Non-blocking, but would be nice to improve.

lng-stripe commented on paymentsheet/src/test/java/com/stripe/android/link/ui/wallet/LinkInline2FASectionScreenshotTest.kt:
```diff
@@ -0,0 +1,162 @@
+        }
+    }
+
+    @Test
+    fun testWithPaymentDetailsMastercard() {
```
Can you add a test for bank account?

lng-stripe reviewed (changes requested):
Mostly small stuff, except I am concerned about the bank account UI.

carlosmuvi-stripe commented:
@lng-stripe addressed feedback here - https://github.com/stripe/stripe-android/pull/11171/commits/2f6ed537b8dadc81a4b7a9cccc9a87cc5566995f

- Removed `setAccountNullable` and updated tests
- Reused bank icon logic from wallet
- Added screenshot tests for banks
